src/Core/Composer/ModulesConfigurationProcessor.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * 2007-2019 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Composer;
+
+use Composer\IO\IOInterface;
+use Symfony\Component\Process\Process;
+use Symfony\Component\Process\Exception\ProcessFailedException;
+
+final class ModulesConfigurationProcessor
+{
+    /**
+     * @var IOInterface the CLI IO interface
+     */
+    private $io;
+
+    public function __construct(IOInterface $io)
+    {
+        $this->io = $io;
+    }
```

Comment:
Maybe you should use a `LoggerInterface` instead of `IOInterface`
It is more generic, can be initialized with a `NullLogger` by default thus you can test this class more easily and be used in a more generix scope than a composer event, or cli command

Reply:
As it's already type hinted to an interface I don't see how changing this would make the tests easier, but it definitively would make the implementation more complex :/

See my previous command: our only scope is doing a Composer extension.

Reply:
fair enough!